svea_vision/scripts/object_detect.py

```diff
@@ -143,33 +142,40 @@ def callback(self, image):
         frame = cv2.resize(frame, (self.IMAGE_WIDTH, self.IMAGE_HEIGHT))
         frame = cv2.cvtColor(frame, cv2.COLOR_BGRA2RGB)
 
-        if self.ONLY_OBJECTS:
-            result = self.model.predict(
-                frame,
-                conf=0.5,
-                verbose=False,
-                classes=list(map(self.label_to_class, self.ONLY_OBJECTS)),
-            )[0]
-        else:
-            result = self.model.predict(frame, conf=0.5, verbose=False)[0]
+        result = self.model.track(frame, **self.detect_kwargs)[0]
 
-        result = result.cpu().numpy()
-        boxes = result.boxes.xyxy
-        conf = result.boxes.conf
-        cls = result.boxes.cls
+        # if enabled, modify frame (add bounding boxes)
+        if self.ENABLE_BBOX_IMAGE:
+            frame = result.plot()
+
+        if self.ENABLE_BBOX_IMAGE:
+            new_image = Image()
+            new_image.header = image.header
+            new_image.height = frame.shape[0]
+            new_image.width = frame.shape[1]
+            new_image.encoding = "rgb8"
+            new_image.step = frame.size // new_image.height
+            new_image.data = frame.tobytes()
+
+            self.pub_bbox_image.publish(new_image)
 
-        ## Update tracker
+        if len(result.boxes) == 0 or not result.boxes.is_track:
```

Comment:
This should either be otherway around or only `len(result.boxes) == 0`, depending on the behavior of `len(result.boxes)`. In case the latter throws...